lib/travis/build/git/clone.rb

```diff
@@ -17,6 +17,53 @@ def apply
         end
 
         private
+          TRACE_COMMAND_GIT_TRACE = "GIT_TRACE=true"
+          TRACE_COMMAND_STRACE = "strace"
+          DEFAULT_TRACE_COMMAND = TRACE_COMMAND_GIT_TRACE
+
+          def repo_slug
+            data.repository[:slug].to_s
+          end
+
+          def owner_login
+            repo_slug.split('/').first
+          end
+
+          def trace_git_commands_owners
+            ENV["TRACE_GIT_COMMANDS_OWNERS"].to_s.split(',')
+          end
+
+          def trace_git_commands_slugs
+            ENV["TRACE_GIT_COMMANDS_SLUGS"].to_s.split(',')
+          end
+
+          def trace_git_commands?
+            trace_git_commands_slugs.include?(repo_slug) || trace_git_commands_owners.include?(owner_login)
+          end
+
+          def trace_command
+            if ENV["TRACE_COMMAND"].to_s == TRACE_COMMAND_STRACE
+              TRACE_COMMAND_STRACE
+            else
+              DEFAULT_TRACE_COMMAND
+            end
+          end
+
+          def git_clone
+            if trace_git_commands?
+              sh.cmd "#{trace_command} git clone #{clone_args} #{data.source_url} #{dir}", assert: false, retry: true
```

Comment:
I think you can simply this logic by combining the `git` command into a method:

```ruby
def git_cmd
  trace_git_commands? ? "#{trace_command} git" : "git"
end

def git_clone
  sh.cmd "#{git_command} clone …" , …
end

def git_fetch
  sh.cmd "#{git_command} -C #{dir} …", …
end

def fetch_ref # maybe we want to rename this method as `git_fetch_ref` for consistency
  sh.cmd "#{git_command} fetch …", …
end
```

Reply:
Nice! Fixed.